enhancements/audit-command.md

```diff
@@ -0,0 +1,594 @@
+    - check if they are following the recommendations and conventions defined
+    - check all warnings and errors which will be raised by using the SDK features scorecard and bundle validate
+- Be able to extract a report with the audit results and in some formats such as json. 
+- Be able to run specific custom checks for each bundle in order to audit the catalog
+
+### Non-Goals
+
+- Add new checks or validators. This proposal has the goal to define the command and operation only. The checks should be addressed in their respective proposals.
+- Allow users select what specific validators will be used by the SDK commands. However, it can be addressed in the future.  
+ 
+## Proposal
+
+Add a new command `--audit` with the purpose to audit all desired operator bundle data of a specific index catalog. 
+
+By default, the command output a report where it is possible to have a view of the main characteristics and aspect of all operator bundles shipped in a specified OLM index catalog as check if they are passing in criteria and checks required to ensure their quality. However, accordingly with the purpose of its usage is also possible customize what should or not be done by this operation. 
+
+This command will perform checks by using the [SDK](https://github.com/operator-framework/operator-sdk) available sub commands; `bundle validate` and `scorecard bundle`.
+
+In this way, opm binary will provide the alpha audit sub-command which would like: 
+
+```
+$ alpha --audit --help
+
+Flags:
+  -h, --help                     help for audit
+  -i, --index-image              index image to get package from. 
+  -c, --container-tools          the container tool used to login in the registry and pull and extract the images. One of: [docker, podman] (default "docker")
+  -w, --wait-time duration       minutes to wait for the report to be complete. Example: 30m (default 20m)                    
+  -o, --output                   output format for results. Valid values: xlsx, json (default "xlsx")
+  -s, --scorecard bool           set false to disable the scorecard checks. (default true). Note that scorecard test requires a cluster running. 
+  -v, --validation bool          set false to disable the bundle validation checks. (default true). 
+  -d, --data                     set false to disable the data report. It is useful for who would like to audit the index via only run the bundle and scorecard checks. (default true).
+  -h, --head-only                set the flag to audit only the operator bundles which are the head of the channel. (default all bundles of the index catalog image will be used)
+  -w, --warning                  set false to disable the report of warning(s) found in the checks (default warning and errors will be reported)
+```
+
+Note that the command will:
+- Use the index image to gathering the data
+- Download the operator bundles
+- Use SDK commands to run the checks 
```

Comment:
The first open question raise where we should add the audit. So, we are not saying here that it should be in OPM. PS.: After many thoughts, I do not sure that we should add it to opm at all either

- SDK provides the features required for we check the operator bundles against the criteria. 
- The criteria checks are actually implemented in operator-framework/api and available for any project/tool
- We do not need to re-implement all SDK features to achieve the audit goal and we can only use it and indeed ensure that we are doing the same tests which are provided by the users as well. 

Is it make sense?

Reply:
@jmccormick2001 I agree I don't like coupling the two commands. IF we plan to share logic, then I think we should pull some code out and make it into a library to share.

Reply:
@jmrodri , the Validators rules used by SDK are already in the operator-frameowork/api as the scorecard config. However, IMO we do not need to re-implemented anything that already exists in the tools. 

See that the audit command ONLY uses SDK binary as any person/job/tool can do so. However, do you see any reason for we re-implement SDK features in another tool/place and have the effort to keep more than once place maintained and etc instead of just call the bin for the audit command in the alpha stage?

Reply:
Really thank you for all your inputs. I am closing this one because I think could address in the EP. However, please feel free to ping me if you see that it requires change.